daemon/image_delete.go

```diff
@@ -88,7 +88,7 @@ func (daemon *Daemon) DeleteImage(eng *engine.Engine, name string, imgs *engine.
 		return nil
 	}
 
-	if len(repos) <= 1 {
+	if len(repos) <= 1 || repoName == "" {
```

Comment:
Considering changing this to `len(repos) - len(tags) == 0` because really, what this question should be asking, is whether or not there will be any tags attached to the image after removing the requested tags.